src/sql/src/plan/query.rs

```diff
@@ -3070,6 +3085,53 @@ fn plan_subscript_slice(
     .into())
 }
 
+fn plan_like(
+    ecx: &ExprContext,
+    expr: &Expr<Aug>,
+    pattern: &Expr<Aug>,
+    escape: Option<&Expr<Aug>>,
+    case_insensitive: bool,
+    not: bool,
+) -> Result<HirScalarExpr, PlanError> {
+    let ecx = ecx.with_name("LIKE argument");
+    let expr = plan_expr(&ecx, expr)?;
+    let haystack_type = ecx.scalar_type(&expr);
+    let haystack = match haystack_type {
+        Some(ScalarType::Char { length }) => HirScalarExpr::CallUnary {
+            func: UnaryFunc::PadChar(expr_func::PadChar { length }),
+            expr: Box::new(expr.type_as(&ecx, &haystack_type.unwrap())?),
+        },
+        _ => expr.type_as(&ecx, &ScalarType::String)?,
+    };
```

Comment:
You can use a name binding in the pattern to avoid the unwrap here:

```suggestion
    let haystack = match ecx.scalar_type(&expr) {
        Some(ref ty @ ScalarType::Char { length }) => HirScalarExpr::CallUnary {
            func: UnaryFunc::PadChar(expr_func::PadChar { length }),
            expr: Box::new(expr.type_as(&ecx, ty)?),
        },
        _ => expr.type_as(&ecx, &ScalarType::String)?,
    };
```